pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/PredicateUtils.java

```diff
@@ -139,10 +143,40 @@ public static IntSet getDictIdSet(BaseInPredicate inPredicate, Dictionary dictio
         }
         break;
       case STRING:
-        for (String value : values) {
-          int dictId = dictionary.indexOf(value);
-          if (dictId >= 0) {
-            dictIdSet.add(dictId);
+        if (values.size() > RequestContextUtils.THRESHOLD_OF_LARGE_IN_CLAUSE_VALUES) {
```

Comment:
I think this should consider the ratio of the size to the dictionary cardinality. What if you have 1001 strings in the in clause but 100000 in the dictionary? Binary search is still going to win here.

Reply:
Also, take a look at galloping intersection here https://github.com/RoaringBitmap/RoaringBitmap/blob/fd54c0a100629bb578946e2a0bf8b62784878fa8/RoaringBitmap/src/main/java/org/roaringbitmap/Util.java#L1028

Reply:
Yes, need more benchmark around this number.

Reply:
I was thinking if we can still benefit from the strings being sorted in this scenario if we use something like https://github.com/apache/pinot/pull/10254#discussion_r1106747320?